src/engine/Server.cpp

```diff
@@ -394,6 +395,12 @@ Awaitable<void> Server::process(
     logCommand(cmd, "clear cache completely (including unpinned elements)");
     cache_.clearAll();
     response = createJsonResponse(composeCacheStatsJson(), request);
+  } else if (auto cmd = checkParameter("cmd", "clear-delta-triples")) {
+    requireValidAccessToken("clear-delta-triples");
+    logCommand(cmd, "clear delta triples");
+    index_.deltaTriplesManager().clear();
```

Comment:
This command (as well as the handling of Update requests)
might take a long time (as it has to wait for other updates to finish). In particular it stalls a server thread.
So :
have a look at the `computeInNewThread`, implement it in a way that makes the pool configurable, and then create a single-threaded pool for the Updates, and shedule anything update-related which is expensive (in particular everything that takes the mutex locks inside the deltaTriplesManager) onto that thread, so that we don't deadlock queries by Updates.